### PR TITLE
Added copy function before deletion

### PR DESCRIPTION
**Pull request by Gabisonfire**
Added a preference that when set will copy the file to the set directory before deleting.
If left undefined, will simply delete as usual.

### CODE REVIEW

File: docs-importer/env.sh [hunk 1]
```diff
@@ -6,4 +6,5 @@ sed -i "s/env3/$TEEDY_LANG/g" $file
 sed -i "s,env4,$TEEDY_URL,g" $file
 sed -i "s/env5/$TEEDY_USERNAME/g" $file
 sed -i "s/env6/$TEEDY_PASSWORD/g" $file
+sed -i "s/env7/$TEEDY_COPYFOLDER/g" $file
```
Comment (Cornelicorn) [inline docs-importer/env.sh:9]: Replace the `/` with `,` like: `sed -i "s,env7,$TEEDY_COPYFOLDER,g" $file
`

File: docs-importer/main.js [hunk 2]
```diff
@@ -400,6 +400,10 @@ const importFile = (file, remove, resolve) => {
         }
         spinner.succeed('Upload successful for ' + file);
         if (remove) {
+          if (prefs.importer.copyFolder) {
+            let filename = file.replace(/^.*[\\\/]/, '');
```
Comment (Cornelicorn) [inline docs-importer/main.js:404]: Why declare filename again? Already declared in L331

File: docs-importer/README.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ For the next start, pass the `-d` argument to skip the wizard:
 
 ## Daemon mode
 
-The daemon mode scan the input directory every 30 seconds for new files. Once a file is found and imported, it is **deleted**.
+The daemon mode scan the input directory every 30 seconds for new files. Once a file is found and imported, it is **deleted**. You can set the "copyFolder" value in the preference file to copy the file before deletion.
```
Comment (Cornelicorn) [inline docs-importer/README.md:28]: It would be nice if the daemon would also ask for it.

**Comment by Gabisonfire** [comment]
I didn't add `TEEDY_COPYFOLDER` in the because it would change the default behavior and this is probably only useful in not a lot of cases like mine.

Use case for me is that I like having a raw copy of the documents elsewhere as a fallback. Having an option not to delete was also considered but it was better for me to keep the import folder clean.

File: docs-importer/README.md (hunk 3 above)
Comment (Gabisonfire) [reply docs-importer/README.md:28]: I didn't add it since I thought it should not really be a default option.

**Comment by Gabisonfire** [comment]
> Deletion wouldn't work anyway - my fault - as it would import every file again every 30 seconds.

Thought so too, but I did not read the whole code in depth so I was unsure.

I see your point with consistency, I'll add it and push.

File: docs-importer/main.js [hunk 4]
```diff
@@ -241,11 +241,30 @@ const askLang = () => {
...
     });
   });
 };
 
+const askCopyFolder = () => {
+  console.log('');
+
+  inquirer.prompt([
+    {
+      type: 'confirm',
```
Comment (Cornelicorn) [inline docs-importer/main.js:254]: Should be input

File: docs-importer/main.js [hunk 5]
```diff
@@ -241,11 +241,30 @@ const askLang = () => {
...
+      name: 'copyfolder',
+      message: 'Enter a path to copy files before they are deleted. (Leave empty to disable)',
+      default: prefs.importer.addtags = ""
```
Comment (Cornelicorn) [inline docs-importer/main.js:257]: also should be `prefs.import.copyfolder`

File: docs-importer/README.md (hunk 3 above)
Comment (Cornelicorn) [inline docs-importer/README.md:28]:
```suggestion
The daemon mode scan the input directory every 30 seconds for new files. Once a file is found and imported, it is **deleted**. You can set the a `copyFolder` to copy the file to before deletion.
```

File: docs-importer/README.md [hunk 6]
```diff
@@ -25,7 +25,7 @@ For the next start, pass the `-d` argument to skip the wizard:
 
 ## Daemon mode
 
-The daemon mode scan the input directory every 30 seconds for new files. Once a file is found and imported, it is **deleted**.
+The daemon mode scan the input directory every 30 seconds for new files. Once a file is found and imported, it is **deleted**. You can set the a `copyFolder` to copy the file to before deletion.
```
Comment (Cornelicorn) [inline docs-importer/README.md:28]:
```suggestion
The daemon mode scan the input directory every 30 seconds for new files. Once a file is found and imported, it is **deleted**. You can set a `copyFolder` to copy the file to before deletion.
```

File: docs-importer/main.js [hunk 7]
```diff
@@ -241,11 +241,52 @@ const askLang = () => {
...
+      }
+
+      fs.access(answers.copyFolder, fs.W_OK | fs.R_OK, (error) => {
+        if (error) {
+          spinner.fail('This directory is not writable');
+          askCopyFolder();
+          return;
+        }
+
+        spinner.succeed(answers.copyFolder + ' set!');
```
Comment (Cornelicorn) [inline docs-importer/main.js:282]:
```suggestion
        spinner.succeed('Copy folder set!');
```

File: docs-importer/main.js [hunk 8]
```diff
@@ -241,11 +241,52 @@ const askLang = () => {
...
 
+const askCopyFolder = () => {
+  console.log('');
+
+  inquirer.prompt([
+    {
+      type: 'input',
+      name: 'copyFolder',
+      message: 'Enter a path to copy files before they are deleted or leave empty to disable.',
+      default: prefs.importer.copyFolder = ""
```
Comment (Cornelicorn) [inline docs-importer/main.js:257]:
```suggestion
      default: prefs.importer.copyFolder
```

File: docs-importer/main.js [hunk 9]
```diff
@@ -241,11 +241,52 @@ const askLang = () => {
...
+    }
+  ]).then(answers => {
+    // Save path
+    prefs.importer.copyFolder = answers.copyFolder;
+
+    // Test path
```
Comment (Cornelicorn) [inline docs-importer/main.js:263]: The path should only be tested if `prefs.importer.copyFolder` is set.

File: docs-importer/main.js (hunk 8 above)
Comment (Gabisonfire) [reply docs-importer/main.js]: Sorry, I almost never write JS

File: docs-importer/main.js (hunk 7 above)
Comment (Cornelicorn) [inline docs-importer/main.js]:
```suggestion
        spinner.succeed('Copy folder set!');
```

File: docs-importer/main.js (hunk 8 above)
Comment (Cornelicorn) [inline docs-importer/main.js]:
```suggestion
      default: prefs.importer.copyFolder
```

File: docs-importer/main.js [hunk 10]
```diff
@@ -400,6 +442,9 @@ const importFile = (file, remove, resolve) => {
         }
         spinner.succeed('Upload successful for ' + file);
         if (remove) {
+          if (prefs.importer.copyFolder) {
+            fs.copyFileSync(file, prefs.importer.copyFolder + filename);
+          }
           fs.unlinkSync(file);
         }
```
Comment (Cornelicorn) [inline docs-importer/main.js:449]: This was somewhat right at the beginning, as the tags are removed from the filename
```suggestion
          if (prefs.importer.copyFolder) {
            fs.copyFileSync(file, prefs.importer.copyFolder + file.replace(/^.*[\\\/]/, ''));
            fs.unlinkSync(file);
          }
          else {fs.unlinkSync(file);}
        }
```

File: docs-importer/main.js [hunk 11]
```diff
@@ -241,11 +241,53 @@ const askLang = () => {
...
+          spinner.fail('This directory is not writable');
+          askCopyFolder();
+          return;
+        }
+        spinner.succeed('Copy folder set!');              
+      });
+    });
+  }
+    askDaemon();
+  });
```
Comment (Cornelicorn) [inline docs-importer/main.js:287]:
```suggestion
        spinner.succeed('Copy folder set!');
        askDaemon();              
      });
    });
  }
  else {askDaemon();}
  });
```

File: docs-importer/main.js [hunk 12]
```diff
@@ -241,11 +241,53 @@ const askLang = () => {
...
 };
 
+const askCopyFolder = () => {
+  console.log('');
+
+  inquirer.prompt([
+    {
+      type: 'input',
+      name: 'copyFolder',
+      message: 'Enter a path to copy files before they are deleted or leave empty to disable.',
```
Comment (Cornelicorn) [inline docs-importer/main.js:256]:
```suggestion
      message: 'Enter a path to copy files before they are deleted or leave empty to disable. The path must end with a `/` on MacOS and Linux or with an `\` on Windows.',
```

File: docs-importer/main.js [hunk 13]
```diff
@@ -241,11 +241,54 @@ const askLang = () => {
...
 };
 
+const askCopyFolder = () => {
+  console.log('');
+
+  inquirer.prompt([
+    {
+      type: 'input',
+      name: 'copyFolder',
+      message: 'Enter a path to copy files before they are deleted or leave empty to disable. The path must end with a `/` on MacOS and Linux or with an `\` on Windows.',
```
Comment (Cornelicorn) [inline docs-importer/main.js:256]:
```suggestion
      message: 'Enter a path to copy files before they are deleted or leave empty to disable. The path must end with a \'/\' on MacOS and Linux or with a \'\\\' on Windows.',
```

**Comment by Cornelicorn** [comment]
Sorry for the mess. I hope you move this to another pull request as I would also like to use this feature. But it has to stay optional and be configurable easily. And it of course should work. Please don't feal discouraged by my sloppy commenting. @Gabisonfire

**Comment by Gabisonfire** [comment]
No worries, it's really not about your comments at all. I'm really a hobbyist coder and never wrote JS before, I thought it would be easier but I'll let someone else do the work. Thanks for your input and sorry for the time spent.

**Comment by Cornelicorn** [comment]
> No worries, it's really not about your comments at all. I'm really a hobbyist coder and never wrote JS before, I thought it would be easier but I'll let someone else do the work. Thanks for your input and sorry for the time spent.

Well you're nearly there...
I also learnt JS with a friend when I first opened the pull request fort the importer last week. Open Source is also about learning coding and improving others code. The last bit could be implemented by changing L261 to `prefs.importer.copyFolder = answers.copyFolder=='undefined' ? '' : answers.copyFolder;`. If you type in `undefined` this will return '' and the copy feature will be disabled.
Then just change L256 to something like:
`message: 'Enter a path to copy files before they are deleted or leave empty to disable. The path must end with a \'/\' on MacOS and Linux or with a \'\\\' on Windows. Entering \'undefined\' will disable this again after setting the folder.',`

**Comment by Gabisonfire** [comment]
https://github.com/sismics/docs/pull/418

**Pull request closed without merging**